third_party/nlbuild-autotools/repo/autoconf/m4/ax_check_compiler.m4

```diff
@@ -31,8 +31,16 @@ AC_DEFUN([_AX_CHECK_COMPILER_OPTION_WITH_VAR],
     AC_MSG_CHECKING([whether the _AC_LANG compiler understands $3])
     SAVE_[]_AC_LANG_PREFIX[]FLAGS=${_AC_LANG_PREFIX[]FLAGS}
     SAVE_$2=${$2}
-    _AC_LANG_PREFIX[]FLAGS=$3
+    AS_LITERAL_IF([$3],
```

Comment:
The upstream for `third_party/nlbuild-autotools` is https://github.com/nestlabs/nlbuild-autotools. Could you try submitting these changes to upstream first?

Reply:
Sure, I'll revert this commit and create a pull request to  https://github.com/nestlabs/nlbuild-autotools first.

I added these changes because without them gcc4 is fed the -Wno-c++14-compat flag, which it doesn't recognize, so CI fails when there are undefined macros (due to -Werror). gcc > 4 accepts this switch.